### ISSUE TITLE
blitCopyMask() support for wDstX < wSrcX

### ISSUE THREAD

**Issue by approxit** [AmigaPorts/ACE#162]
It finally happened. After around 7 years, we just got use case for blitting with wDstX < wSrcX, which currently is not implemented.

In https://github.com/Last-Minute-Creations/villages/commit/77b6cd046187d4ae020b38b01df7a20b34d7b6bd I've prepared bugged version of blitting (hit `space` to actually render bitmap). In this use case, I'm trying to slide a bitmap from offscreen left side to center. Because we don't want to blit outside of buffer, we need to draw smaller chunks of bitmap. Details shown in below picture, made with such a professional tool:

![obraz](https://user-images.githubusercontent.com/6125496/195663087-041e1c21-5d66-46a3-9f68-272e0f5140b3.png)

**Comment by tehKaiN** [comment]
This requires blitting in descending mode so that pixels are blitted from bottom-right corner instead of top-left one as well as it will reverse the pixel shifting direction from right to left.

**Comment by Vairn** [comment]
Did I fix this. or is this something different?

**Comment by tehKaiN** [comment]
afaik it's still there - and the title is slightly wrong, because only `blitCopyMask()` has this shortcoming, `blitCopy()` is fine. It was more or less intentional to make it so in order for that function to be a bit faster calculation-wise.

Now that we have dedicated bob system, this shortcoming can be fixed since it's no longer the fastest way to draw masked objects and could be more universal.

**Comment by tehKaiN** [comment]
Yeah, my memory tricked me, apparently it's fixed in https://github.com/AmigaPorts/ACE/commit/c6824b2f9228f7a37e85b30bf67ecd9ded25784e

it would be nice to test the mentioned code though, will try to do it today.

**Comment by tehKaiN** [comment]
Haven't found any issues in mentioned code, not sure if I reproduced it properly though.

**Comment by Vairn** [comment]
Well, it basically didn't let half my gfx draw without the changes, since I was copying data from random spots in a texture atlas, to like left side of screen etc.

So it fixed the issues I had.
